File: src/main/java/com/wesker926/leetcode/algorithms/p0319/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/15
// @description: 只有平方数会亮
package s1

import "math"

func bulbSwitch(n int) int {
	// 加0.5为了避免精度问题
	return int(math.Sqrt(float64(n) + 0.5))
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0319/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0319.s1;

/**
 * @author wesker.gh
 * @date 2021/11/15
 * @description 只有平方数会亮
 */
public class Solution {
    public int bulbSwitch(int n) {
        // 加0.5为了避免精度问题
        return (int) Math.sqrt(n + 0.5);
    }
}
